File: Arquivo.java
import java.io.RandomAccessFile;
import java.io.IOException;

class Arquivo {

    public static void main(String args[]) throws Exception{  //jogando excecao
        
        int numero = MyIO.readInt();
        
        RandomAccessFile file = new RandomAccessFile("data.txt", "rw");  //criando a class RandomAcessFile na class Arquivo e
                                                                                  //abrindo "data.txt" no modo de ler e escrever 'rw'


        for (int i = 0; i < numero; i++) 
        {
            double valor = MyIO.readDouble();       //le o valor e salva em 'valor'
            file.writeDouble(valor);               //escreve o valor no arquivo
        }

        file.close();



        file = new RandomAccessFile("data.txt", "r");  //abrindo novamente o arquivo, mas no modo de leitura 'r'

        for (int j = numero - 1; j >= 0; j--) 
        {
            file.seek(j * 8);     //cada numero é um double, ou seja, ocupa 8 bytes. é necessario multiplicar j por 8 para obter o inicio de cada valor
            double valor2 = file.readDouble();   //atribuindo a valor2 o valor lido no arquivo
            
            if(valor2 == (int)valor2)         //caso valor real seja igual ao valor inteiro (.0), imprimir o valor de forma inteira para 
                MyIO.println((int) valor2);                                                          //evitar printar 10.0 por exemplo
            else
                MyIO.println(valor2);   //caso numero for real, mostrar com ponto '.'
        }
    
        
        file.close();    
        
        
    }
}

File: Arquivo.c
#include <stdbool.h>
#include <stdio.h>
#include <string.h>


int main() {

  int n;
  scanf("%d", &n);
  
  double valorLido, valor;
  float valor2[1000];

  FILE *arq = fopen("arq.txt", "w");  //abrindo arquivo no modo de escrita 'w'

  for (int i = 0; i < n; i++) 
  {
    scanf(" %lf", &valorLido);          //lendo valor 
    fprintf(arq, " %f", valorLido);    //salvando valor lido no arquivo 'arq'
  }

  fclose(arq);


  
  arq = fopen("arq.txt", "r");      //abrindo novamente o arquivo, mas no modo de leitura 'r'

  int tam = 0;             //tamanho do array
  while (fscanf(arq, " %f", &valor2[tam]) == 1) {   //calcular numero de numeros no arquivo 
        tam++;            //vai aumentando até que tenham valores no array
  }

  
  for (int i = tam -1; i >= 0; i--) 
  {
    
    fscanf(arq, " %f", &valor2[i]);   //lendo o valor do arquivo e salvando em valor2

    if(valor2[i] == (int)valor2[i])
      printf("%d\n", (int)valor2[i]);
    else 
      printf("%.3f\n", valor2[i]);        //mostrando na tela o valor2 - que foi lido do arquivo

  }

  fclose(arq);

  return 0;
}


